### PR TITLE
Fixes things I broke in the Transducer with single-embeddings design …

### PR DESCRIPTION
**Pull request by Adamits**
…and when updating inference.

### CODE REVIEW

File: yoyodyne/models/transducer.py [hunk 1]
```diff
@@ -38,12 +38,10 @@ def __init__(
...
         """
-        # Gets number of non-target symbols.
-        source_vocab_size = kwargs["vocab_size"] - kwargs["target_vocab_size"]
-        # This is the size of the shared embedding matrix.
-        # It must contain every possible source AND target symbol.
-        kwargs["vocab_size"] = source_vocab_size + len(expert.actions)
-        # Alternate outputs than dataset targets.
+        self.initial_vocab_size = kwargs["vocab_size"]
+        kwargs["vocab_size"] += len(expert.actions)
+        self.initial_vocab_size
```
Comment (kylebgorman) [inline yoyodyne/models/transducer.py:43]: no-op line.

File: yoyodyne/models/transducer.py [hunk 2]
```diff
@@ -38,12 +38,10 @@ def __init__(
             *args: passed to superclass.
             **kwargs: passed to superclass.
         """
-        # Gets number of non-target symbols.
-        source_vocab_size = kwargs["vocab_size"] - kwargs["target_vocab_size"]
-        # This is the size of the shared embedding matrix.
-        # It must contain every possible source AND target symbol.
-        kwargs["vocab_size"] = source_vocab_size + len(expert.actions)
-        # Alternate outputs than dataset targets.
+        self.initial_vocab_size = kwargs["vocab_size"]
```
Comment (kylebgorman) [inline yoyodyne/models/transducer.py:41]: I think a comment explaining what "initial" means would be useful here.

**Review by kylebgorman — COMMENTED** [review]
Just took a quick look.

**Comment by bonham79** [comment]
This won't fix the issue. When you merged the vocab sizes, you didn't update the decoders to produce outputs to aligned to target vocab size. They all instead have embeddings linked to total vocab size. To fix the transducer issue you need to first update all the upstream models so the target vocab is lining up.

**Comment by kylebgorman** [comment]
What's the status of this @Adamits?

**Comment by Adamits** [comment]
This was copied/merged into https://github.com/CUNY-CL/yoyodyne/pull/247. we can close it.

**Pull request closed without merging**